dfetch/commands/list.py

```diff
@@ -0,0 +1,46 @@
+"""List metadata of all projects listed in the manifest.
+
+For each project listed in the manifest the metadata is shown,
+such as timestamp of last fetch, revision, remote url, etc.
+"""
+
+import argparse
+import os
+
+import dfetch.commands.command
+import dfetch.manifest.manifest
+from dfetch.log import get_logger
+from dfetch.project.metadata import Metadata
+
+logger = get_logger(__name__)
+
+
+class List(dfetch.commands.command.Command):
+    """List metadata for each project.
+
+    List all information provided in manifest and metadata files.
+    """
+
+    @staticmethod
+    def create_menu(subparsers: "argparse._SubParsersAction") -> None:
+        """Add the parser menu for this action."""
+        dfetch.commands.command.Command.parser(subparsers, List)
+
+    def __call__(self, args: argparse.Namespace) -> None:
+        """Perform the list."""
+        manifest, path = dfetch.manifest.manifest.get_manifest()
+
+        with dfetch.util.util.in_directory(os.path.dirname(path)):
+            for project in manifest.projects:
+                logger.print_info_line("project", project.name)
+                logger.print_info_line("    remote", project.remote)
+                try:
+                    metadata = Metadata.from_file(
+                        Metadata.from_project_entry(project).path
+                    )
+                    logger.print_info_line("    remote url", metadata.remote_url)
+                    logger.print_info_line("    branch", metadata.branch)
```

Comment:
When this was implemented, only `branch` existed, but there currently is also a possiblity that `tag:` exists. Maybe in one of the feature tests use tag instead of `branch`.

Reply:
Hmm, interesting remark. Should `branch` (and possibly `revision`) be shown when `tag` is present? I presume `branch` will be empty when `tag` is used.

Or an even more general thought, should it show empty fields (for example: patch, remote, ...)

Reply:
Nice solution with the `<none>`